cmake/preproject.cmake

```diff
@@ -63,7 +63,14 @@ endif()
 
 function(add_warning_checks name)
   target_compile_options(
-    ${name} PRIVATE -Wall -Wextra -Werror -Wundef -Wpedantic
+    ${name}
+    PRIVATE -Wall
+            -Wextra
+            -Werror
+            -Wundef
+            -Wpedantic
+            -Wno-unused
```

Comment:
Instead, we could disable those warnings just for the logging code via `#pragma`s (like @jumaffre just did in another PR for a different purpose).

Reply:
I don't think we can do this. The pragma disables warnings in a specific scope, but the unused variable exists at an outer scope from the macro

ie
```cpp
void foo(Widget w)
{
  LOG_DEBUG_FMT("My widget is {}", w);
}
```
becomes
```cpp
void foo(Widget w)
{
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wunused-parameter" ...
// No body nop
#pragma clang diagnostic pop
}
```
and the warning remains on the first line. The `Wgnu-zero-variadic-macro-arguments` is a special case since the warning arises on the macro implementation line, so it can be disabled there rather than at the call-site.

Reply:
You're right, the macro doesn't include that scope, this wouldn't work.